### PR TITLE
Add NewReaderWithOffset

### PR DESCRIPTION
**Pull request by jfontan**
This allows opening siva files while they are being modified or opening an old index.

### CODE REVIEW

File: index.go [hunk 1]
```diff
@@ -462,17 +462,22 @@ func readBinary(r io.Reader, data []interface{}) error {
 	return nil
 }
 
-func readIndex(r io.ReadSeeker) (Index, error) {
-	endLastBlock, err := r.Seek(0, io.SeekEnd)
-	if err != nil {
-		return nil, err
+func readIndex(r io.ReadSeeker, o uint64) (Index, error) {
```
Comment (mcarmonaa) [inline index.go]: Maybe it'd be worthy write a comment to know why `readIndex` and `readIndexAt` are different, right now it's confusing.

File: reader.go [hunk 2]
```diff
@@ -33,6 +34,16 @@ func NewReader(r io.ReadSeeker) Reader {
 	return &reader{r: r}
 }
 
+// NewReaderWithOffset creates a new Reader giving the position of the index.
+// This is useful to open siva files that are being written or reading an
+// old index.
+func NewReaderWithOffset(r io.ReadSeeker, o uint64) Reader {
```
Comment (ajnavarro) [inline reader.go]: what happens if the provided offset is not an index?

File: reader_test.go [hunk 3]
```diff
@@ -126,3 +127,25 @@ func (s *ReaderSuite) testIndexGlob(c *C, pattern string, expected []string) {
 
 	c.Assert(f.Close(), IsNil)
 }
+
+func (s *ReaderSuite) TestOffset(c *C) {
```
Comment (ajnavarro) [inline reader_test.go]: We should test also non-happy paths.

**Comment by jfontan** [comment]
Found a problem that causes index to be read every time it is accessed. Working on a fix.

File: reader.go (hunk 2 above)
Comment (jfontan) [reply reader.go]: Calling `Index` function will error. The same way it errors when opening a broken siva file.

**Pull request merged**